### PR TITLE
Workaround for DS issue in Llama

### PR DESCRIPTION
**Pull request by ugolowic**
DeepSpeed module doesn't handle keyword arguments and throws a RuntimeError when OH calls it with kwargs. This is a workaround for this issue in Llama.

### CODE REVIEW

**Comment by yafshar** [comment]
@ugolowic Could you please add the example that causes the issue to the PR README? This will help us clearly identify the problem and provide a solution if we encounter the same issue with other models.

**Comment by ugolowic** [comment]
@yafshar 
`MASTER_ADDR=localhost MASTER_PORT=12345 PT_HPU_LAZY_MODE=0   python3 examples/gaudi_spawn.py   --world_size 8 --use_deepspeed examples/language-modeling/run_lora_clm.py --dataset_name tatsu-lab/alpaca --bf16 False --dataset_concatenation --max_seq_len 4096 --validation_split_percentage 4 --save_strategy no --warmup_ratio 0.03 --lr_scheduler_type "cosine" --logging_steps 10 --do_train --use_habana --throughput_warmup_steps 3 --use_flash_attention True --flash_attention_causal_mask True --num_train_epochs 1 --eval_delay 2 --lora_rank 4 --lora_target_modules "q_proj" "v_proj" "k_proj" "o_proj" --per_device_eval_batch_size 1 --attn_softmax_bf16 True --evaluation_strategy epoch --dataloader_num_workers 4 --pipelining_fwd_bwd --learning_rate 0.0018 --deepspeed examples/language-modeling/llama3_ds_zero3_4k.json --gradient_accumulation_steps 32 --flash_attention_recompute True --context_parallel_size 2 --per_device_train_batch_size 1 --use_zero3_leaf_promotion --cache_size_limit 256 --compile_dynamic 0 --model_name_or_path meta-llama/Llama-3.3-70B-Instruct --fp8 True --output_dir /tmp/llama2_70b_hf_g2 --minimize_memory --use_lazy_mode False --torch_compile_backend hpu_backend --torch_compile`

**Comment by yafshar** [comment]
@ugolowic, using this changes makes the code order-dependent. I suggest to add the comment for those argument which have different naming.

**Review by yafshar — APPROVED** [review]
LGTM!

@regisss this PR is ready. Would you please do the final check!

**Review by regisss — APPROVED** [review]
LGTM

I guess this happens with other models too then, because they all use keyword arguments in the non-gradient-checkpointing case.
